File: client-app/src/models/Syllabus.ts
import { RcFile } from "antd/es/upload";

export interface Syllabus {
  id: string;
  nameTH: string;
  nameEN: string;
  year: string;
  degreeTH: string;
  degreeEN: string;
  pdf: string;
  subjectGeneral: Subject[];
  subjectSpecific: Subject[];
  subjectFreeElective: Subject[];
  objective: string[];
  occupation: string[];
  createdAt: Date;
  hidden: boolean;
  total: number;
  avg: number;

  subjectGeneralSum?: number;
  subjectSpecificSum?: number;
  subjectFreeElectiveSum?: number;
  pdfPreview?: string;
  yearFormat?: string;
}

export interface SubjectCreate {
  name: string;
  credit: number;
}

export interface Subject extends SubjectCreate {
  id?: string;
  subjectCategory: number;
}

export interface SyllabusCreate {
  nameTH: string;
  nameEN: string;
  year: string;
  pdf: string;
  degreeTH: string;
  degreeEN: string;
  subjects: SubjectCreate[];
  objective: any[];
  occupation: any[]; 

  subjectGeneral: Subject[];
  subjectSpecific: Subject[];
  subjectFreeElective: Subject[];
}

export interface SyllabusDetail {
  id: string;
  nameTH: string;
  nameEN: string;
  year: string;
  degreeTH: string;
  degreeEN: string;
  pdf: string;
  subjectGeneral: Subject[];
  subjectSpecific: Subject[];
  subjectFreeElective: Subject[];
  objective: Obj[];
  occupation: Obj[];
  hidden: boolean;

  yearPreview?: string;
}

interface Obj {
  id: string;
  sentence: string;
}

File: Domain/Others/Syllabus.cs
 
namespace Domain.Others
{
    public class Syllabus
    {
        public string Id { get; set; }  
        public string NameTH { get; set; }
        public string NameEN { get; set; }
        public string Year { get; set; }
        public string DegreeTH { get; set; }
        public string DegreeEN { get; set; } 
        public string PDF { get; set; } 
        public bool IsUsed { get; set; } = true;
        public bool Hidden { get; set; } = false;
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public ICollection<Objective> Objectives { get; set; } = new List<Objective>();
        public ICollection<Occupation> Occupations { get; set; } = new List<Occupation>();
        public ICollection<Subject> Subjects { get; set; } = new List<Subject>();
    }
}